### PR TITLE
#WOB-625 add filter by time off

### PR DESCRIPTION
**Pull request by SergeyPish**

### CODE REVIEW

File: app/src/timer-planning/timer-planning.controller.ts [hunk 1]
```diff
@@ -147,6 +147,12 @@ export class TimerPlanningController {
             return res.status(HttpStatus.FORBIDDEN).json({ message: 'ERROR.CHECK_REQUEST_PARAMS' });
         }
 
+        let timerOffIds = body.timerOffIds;
+
+        if (!body.timerOffIds) {
+            timerOffIds = [];
```
Comment (athlonUA) [inline app/src/timer-planning/timer-planning.controller.ts]: Extra coding. Use simplest solution:
`const timerOffIds = body.timerOffIds || [];`

File: app/src/timer-planning/timer-planning.controller.ts [hunk 2]
```diff
@@ -193,6 +200,12 @@ export class TimerPlanningController {
...
```
Comment (athlonUA) [inline app/src/timer-planning/timer-planning.controller.ts]: The same issue. See above

File: app/src/timer-planning/timer-planning.service.ts [hunk 3]
```diff
@@ -235,8 +242,17 @@ export class TimerPlanningService {
...
+        timerOffIds: string[]
     ): Promise<AxiosResponse | AxiosError> {
+        let timerOffIdsString = ``;
+        if (timerOffIds.length) {
+            timerOffIdsString = `
+                timer_off_id: {
+                    _in: [${timerOffIds.map((id: string) => `"${id}"`).join(',')}]
+                },
+            `;
+        }
```
Comment (athlonUA) [inline app/src/timer-planning/timer-planning.service.ts]: Prettify code style: add a new line after the close bracket

File: app/src/timer-planning/timer-planning.service.ts [hunk 4]
```diff
@@ -235,8 +242,17 @@ export class TimerPlanningService {
         startDate: string,
         endDate: string,
         userIds: string[],
-        teamId: string
+        teamId: string,
+        timerOffIds: string[]
     ): Promise<AxiosResponse | AxiosError> {
+        let timerOffIdsString = ``;
```
Comment (athlonUA) [inline app/src/timer-planning/timer-planning.service.ts]: Rename to `timerOffIdsExpression`

File: app/src/timer-planning/timer-planning.service.ts [hunk 5]
```diff
@@ -257,6 +273,7 @@ export class TimerPlanningService {
                     },
                     where: {
                         team_id: {_eq: "${teamId}"},
+                        ${timerOffIdsString}
```
Comment (athlonUA) [inline app/src/timer-planning/timer-planning.service.ts]: add coma at the end of the line

Comment (SergeyPish) [reply app/src/timer-planning/timer-planning.service.ts]: there is a comma in the expression itself

**Pull request merged**